plugin/src/main/java/com/iridium/iridiumcore/utils/Placeholder.java

```diff
@@ -32,15 +37,34 @@ public Placeholder(String key, String value) {
      * @param value The actual value of the placeholder
      */
     public Placeholder(String key, Supplier<String> supplier) {
-        this.key = "%" + key + "%";
+        this.key = key;
+        this.formattedKey = "%" + key + "%";
         this.supplier = supplier;
     }
 
-    public String getValue()
-    {
+    /**
+     * The default constructor.
+     *
+     * @param key   The placeholder without curly brackets.
+     * @param value The actual value of the placeholder
+     */
+    public Placeholder(String key, Function<Integer, String> supplier) {
+        this.key = key;
+        this.formattedKey = "%" + key + "%";
+        this.int_supplier = supplier;
+    }
+
+    public String getValue() {
         if(value==null && supplier!=null) value = supplier.get();
-        return value; 
+        return value;
```

Comment:
There is a case where both value and supplier can be null here now no? Is this an issue?

Reply:
It's an issue with normal placeholder it will just return null without throwing